google/resource_compute_instance.go

```diff
@@ -336,11 +327,13 @@ func resourceComputeInstance() *schema.Resource {
 									"nat_ip": &schema.Schema{
 										Type:     schema.TypeString,
 										Optional: true,
+										Computed: true,
 									},
 
 									"assigned_nat_ip": &schema.Schema{
-										Type:     schema.TypeString,
-										Computed: true,
+										Type:       schema.TypeString,
+										Computed:   true,
+										Deprecated: "Please use nat_ip",
```

Comment:
What is the benefit of changing the way people use the nat ip? This PR is already big, I would leave this change out of it for now and open an issue (explaining why it is desirable and how the benefits outweigh having to force users to change their config) if you feel strongly about it.

Note: I don't know why they where split in two fields in the first place, I wasn't there when the decision was taken, maybe @danawillow has more insights.

Reply:
Nah, I wasn't there either. Maybe it can change due to reasons other than setting the field via the API? There's nothing in the docs that gives me that impression though, it seems like if you set it it'll be what you set it to, and if you don't set it it'll assign something. @evandbrown do you know?

Reply:
I guessed that perhaps the original implementor was unaware that a field could be both optional and computed - i.e. could be set by a human or determined by GCE - and thus deduplicated them.

I don't feel strongly about making this change though - happy to leave it out of this PR.

Reply:
I would remove the `Deprecated` notice from `assigned_nat_ip` but you can leave the `nat_ip` field as `Computed` and set the value from GCE. This way, we don't force user to change their config for no clear benefit and the logic for flatten/expand is simpler.

Reply:
SGTM - Will do.

Reply:
Done.